extensions/pkg/predicate/predicate_test.go

```diff
@@ -87,6 +89,72 @@ var _ = Describe("Predicate", func() {
 		})
 	})
 
+	Describe("#AddTypePredicate", func() {
+		var (
+			object       client.Object
+			createEvent  event.CreateEvent
+			updateEvent  event.UpdateEvent
+			deleteEvent  event.DeleteEvent
+			genericEvent event.GenericEvent
+		)
+
+		BeforeEach(func() {
+			object = &extensionsv1alpha1.Extension{
+				Spec: extensionsv1alpha1.ExtensionSpec{
+					DefaultSpec: extensionsv1alpha1.DefaultSpec{
+						Type: extensionType,
+					},
+				},
+			}
+			createEvent = event.CreateEvent{
+				Object: object,
+			}
+			updateEvent = event.UpdateEvent{
+				ObjectOld: object,
+				ObjectNew: object,
+			}
+			deleteEvent = event.DeleteEvent{
+				Object: object,
+			}
+			genericEvent = event.GenericEvent{
+				Object: object,
+			}
+		})
```

Comment:
You could reuse this from the earlier Describe block.

Reply:
Every `Describe` block is using a different object, so we cannot use this block for all `Describe`. We can define a function similar to [this](https://github.com/gardener/gardener/blob/7e8138bd5ad459c022e47339691da54f3d3fe9a3/extensions/pkg/predicate/predicate_test.go#L387) though.